### ISSUE TITLE
Failure test:baseservices\\threading\\monitor\\enter\\monitorenter\\monitorenter.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#55895]
Run [runtime-coreclr r2r-extra 20210717.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1244563&view=ms.vss-test-web.build-test-results-tab&runId=36992504&resultId=104360&paneView=debug)

Failed test:
```
R2R windows arm64 Checked gcstress0xf @ Windows.10.Arm64v8.Open

- baseservices\\threading\\monitor\\enter\\monitorenter\\monitorenter.cmd

```
**Error message:**
```
Could Not Find D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.dll.rsp

Return code:      1
Raw output file:      D:\h\w\A8800925\w\AC58096E\uploads\Reports\baseservices.threading\monitor\enter\monitorenter\monitorenter.output.txt
Raw output:
BEGIN EXECUTION
monitorenter.dll
1 file(s) copied.
Response file: D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.dll.rsp
D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\IL-CG2\monitorenter.dll
-o:D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.dll
--targetarch:arm64
--verify-type-and-field-layout
-r:D:\h\w\A8800925\p\System..dll
-r:D:\h\w\A8800925\p\Microsoft..dll
-r:D:\h\w\A8800925\p\mscorlib.dll
-r:D:\h\w\A8800925\p\netstandard.dll
-O
" "dotnet" "D:\h\w\A8800925\p\crossgen2\crossgen2.dll" @"D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.dll.rsp"   -r:D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\IL-CG2*.dll"
Emitting R2R PE file: D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.dll
"D:\h\w\A8800925\p\corerun.exe"  monitorenter.dll
null object tests
tookLock == true tests
timeout < -1
timeout > int.Max
--- no contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
--- a little contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
--- lots of contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
*** High thread ID tests ***
Assertion failed
--- no contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
--- a little contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
--- lots of contention ---
Positive tests, no SyncBlk
Positive tests, with HashCode
Positive tests, with SyncBlk
Failed!
Expected: 100
Actual: 101
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\A8800925\p
D:\h\w\A8800925\w\AC58096E\e\baseservices\threading\monitor\enter\monitorenter\monitorenter.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._monitor_enter_monitorenter_monitorenter_._monitor_enter_monitorenter_monitorenter_cmd() in baseservices.threading.XUnitWrapper.dll:token 0x60001d2+0x284
```

**Comment by mangod9** [comment]
@kouvel

**Comment by VincentBu** [comment]
Failed again in [runtime-coreclr outerloop 20210728.8](https://dev.azure.com/dnceng/public/_build/results?buildId=1264799&view=ms.vss-test-web.build-test-results-tab&runId=37460660&paneView=debug&resultId=109090)

Failed test:
```
R2R-CG2 windows arm Checked no_tiered_compilation @ Windows.10.Arm64v8.Open

- JIT\\Regression\\VS-ia64-JIT\\V2.0-Beta2\\b311420\\b311420\\b311420.cmd
- JIT\\Regression\\VS-ia64-JIT\\V1.2-M02\\b28158\\b28158\\b28158.cmd
- JIT\\jit64\\opt\\cse\\hugeexpr1\\hugeexpr1.cmd
- JIT\\jit64\\opt\\cse\\HugeField2\\HugeField2.cmd
- Loader\\classloader\\MethodImpl\\CovariantReturns\\UnitTest\\UnitTestMultiModule\\UnitTestMultiModule.cmd
- JIT\\opt\\virtualstubdispatch\\hashcode\\ctest1_cs_ro\\ctest1_cs_ro.cmd
- JIT\\opt\\virtualstubdispatch\\hashcode\\ctest1_cs_do\\ctest1_cs_do.cmd
- JIT\\opt\\virtualstubdispatch\\hashcode\\ctest1_cs_r\\ctest1_cs_r.cmd
- JIT\\opt\\virtualstubdispatch\\hashcode\\ctest1_cs_d\\ctest1_cs_d.cmd
- JIT\\jit64\\valuetypes\\nullable\\box-unbox\\value\\box-unbox-value028\\box-unbox-value028.cmd
- JIT\\jit64\\hfa\\main\\testE\\hfa_nd2E_r\\hfa_nd2E_r.cmd
- Loader\\classloader\\TypeGeneratorTests\\TypeGeneratorTest1478\\Generated1478\\Generated1478.cmd
- JIT\\HardwareIntrinsics\\X86\\Sse2\\StoreHigh_r\\StoreHigh_r.cmd
- JIT\\jit64\\*

```
**One of error messages:**
```
Could Not Find D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.dll.rsp
Stack overflow.
Repeat 319 times:
at System.SR.GetResourceString(System.String)
at System.AccessViolationException..ctor()
at System.SR.InternalGetResourceString(System.String)
at System.SR.GetResourceString(System.String)
at System.AccessViolationException..ctor()
at System.Collections.Generic.List1[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..cctor() at System.SR.InternalGetResourceString(System.String) at System.SR.GetResourceString(System.String) at System.AccessViolationException..ctor() at System.SR..cctor() at System.SR.GetResourceString(System.String) at System.AccessViolationException..ctor() at System.Collections.HashHelpers..cctor() at System.Collections.HashHelpers.GetPrime(Int32) at System.Collections.Generic.Dictionary2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Initialize(Int32)
at System.Collections.Generic.Dictionary2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..ctor(Int32, System.Collections.Generic.IEqualityComparer1<System.__Canon>)
at System.AppContext.Setup(Char**, Char**, Int32)

Return code:      1
Raw output file:      D:\h\w\AFBD0A3C\w\A0150954\uploads\Reports\JIT.Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.output.txt
Raw output:
BEGIN EXECUTION
b311420.dll
1 file(s) copied.
Response file: D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.dll.rsp
D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\IL-CG2\b311420.dll
-o:D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.dll
--targetarch:arm
--verify-type-and-field-layout
-r:D:\h\w\AFBD0A3C\p\System..dll
-r:D:\h\w\AFBD0A3C\p\Microsoft..dll
-r:D:\h\w\AFBD0A3C\p\mscorlib.dll
-r:D:\h\w\AFBD0A3C\p\netstandard.dll
-O
" "dotnet" "D:\h\w\AFBD0A3C\p\crossgen2\crossgen2.dll" @"D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.dll.rsp"   -r:D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\IL-CG2*.dll"
Emitting R2R PE file: D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.dll
"D:\h\w\AFBD0A3C\p\corerun.exe"  b311420.dll
Expected: 100
Actual: -1073741571
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\AFBD0A3C\p
D:\h\w\AFBD0A3C\w\A0150954\e\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b311420\b311420\b311420.cmd
Expected: True
Actual:   False


Stack trace
   at JIT_Regression._VS_ia64_JIT_V2_0_Beta2_b311420_b311420_b311420_._VS_ia64_JIT_V2_0_Beta2_b311420_b311420_b311420_cmd() in JIT.Regression.XUnitWrapper.dll:token 0x6000b94+0x284
```

**Comment by mateoatr** [comment]
This looks like there's an infinite recursion when trying to get the exception message that will be thrown. Probably a dup of https://github.com/dotnet/runtime/issues/53833.

**Comment by kouvel** [comment]
The second failure looks unrelated to the first failure, will track the first failure for this issue

**Comment by davidwrighton** [comment]
This is a dup of #13613, which was fixed with PR #57008. The issue where it failed with a stackoverflow is an unrelated bug which broke all managed code in that run was fixed already as well.
